controllers/akodeploymentconfig/cluster/cluster_controller_addon_secret.go

```diff
@@ -5,25 +5,132 @@ package cluster
 
 import (
 	"context"
+	"errors"
 
 	"github.com/go-logr/logr"
-	akoov1alpha1 "github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/api/v1alpha1"
-	"github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/pkg/ako"
-	akoo "github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/pkg/ako-operator"
+	p "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apiserver/apis/datapackaging/v1alpha1"
+
+	// testutil "github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/pkg/test/util"
+	runv1alpha3 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha3"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	akoov1alpha1 "github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/api/v1alpha1"
+	"github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/pkg/ako"
+	akoo "github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/pkg/ako-operator"
 )
 
+const AkoPackageName = "TestPackage"
+const AkoSecretref = "TestingRef"
+
+func (r *ClusterReconciler) ReconcileClusterBootstrap(
+	ctx context.Context,
+	log logr.Logger,
+	cluster *clusterv1.Cluster,
+	obj *akoov1alpha1.AKODeploymentConfig,
+
+) (ctrl.Result, error) {
+	log.Info("Starts reconciling cluster bootstrap")
+	res := ctrl.Result{}
+	// get the bootstrap
+	bootstrap := &runv1alpha3.ClusterBootstrap{}
+	if err := r.Get(ctx, client.ObjectKey{
+		Name:      cluster.Name,
+		Namespace: cluster.Namespace,
+	}, bootstrap); err != nil {
+		log.Info("Failed to get cluster bootstrap, requeue")
+		return res, err
+	}
+
+	// ------------------------------------------------
+	//  Temporary for testing
+
+	clusterBootstrapPackage := runv1alpha3.ClusterBootstrapPackage{
+		RefName: AkoPackageName,
+		ValuesFrom: &runv1alpha3.ValuesFrom{
+			SecretRef: AkoSecretref,
+		},
+	}
+	// ------------------------------------------------
+
+	bootstrap.Spec.AdditionalPackages = append(bootstrap.Spec.AdditionalPackages, &clusterBootstrapPackage)
+
+	clusterbootstrap := bootstrap.DeepCopy()
+	return res, r.Update(ctx, clusterbootstrap)
+}
+
+var pkgRefName = "load-balancer-and-ingress-service.tanzu.vmware.com"
+
+func getPackage(pl *p.PackageList) (*p.Package, error) {
+	pkg := &p.Package{}
+	for _, n := range pl.Items {
+		if n.Spec.RefName == pkgRefName {
+			pkg = n.DeepCopy()
+			return pkg, nil
+		}
+	}
+	return pkg, errors.New("Could not get AKO package")
+}
+
+// return new CB package list with AKO package removed
+func RemovePackage(cbpl []*runv1alpha3.ClusterBootstrapPackage) ([]*runv1alpha3.ClusterBootstrapPackage, error) {
```

Comment:
make it a bit generic, such as defining `RemovePackageFromCB(clusterBoostrap, package)`

Reply:
consider adding `AddPackageToCB(clusterBoostrap, package)`